Sources/GRButton/GRButton.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  sourcery_iOS
+//
+//  Created by Andrej Jasso on 28/10/2022.
+//
+
+import SwiftUI
+import GoodExtensions
+
+@available(iOS 14.0, *)
+public struct GRButton: View {
+
+    private let viewModel: GRButtonViewModel
+    private let textContent: String?
+    private let isLoading: Bool
+    private let icon: Image?
+    private let action: VoidClosure
+
+    @Environment(\.isEnabled) private var isEnabled
+
+    public init(
+        viewModel: GRButtonViewModel,
+        isLoading: Bool = false,
+        textContent: String?,
+        icon: Image? = nil,
+        action: @escaping VoidClosure
+    ) {
+        self.viewModel = viewModel
+        self.isLoading = isLoading
+        self.textContent = textContent
+        self.icon = icon
+        self.action = action
+    }
+
+    public var body: some View {
+        Button(
+            action: {
+                GRHapticsManager.shared.playSelectionFeedback()
+                action()
+            },
+            label: {
+                ZStack {
+                    if let loading = viewModel.appearance.loading, isLoading {
+                        progress(loading: loading)
+                    } else {
+                        content
+                    }
+                }
+                .frame(maxWidth: !viewModel.appearance.frameModel.stretching ? .none : .infinity)
+                .frame(width: viewModel.appearance.frameModel.width)
+                .frame(height: viewModel.appearance.frameModel.height)
+                .background(background)
+                .shadowed(viewModel.appearance.shadow)
+                .opacity(isEnabled ? 1.0 : 0.4)
+                .animation(.linear(duration: 0.2), value: isLoading)
+            }
+        )
+        .allowsHitTesting(!isLoading)
+    }
+
+    @ViewBuilder
+    private var content: some View {
+        HStack(alignment: .center, spacing: viewModel.appearance.frameModel.interItemSpacing) {
+            if let icon = viewModel.appearance.icon {
+                switch icon.alignment {
+                case .trailingHugged:
+                    if icon.mirrorIconSpace {
+                        emptyIcon
+                    }
+                    text
+                    self.icon
+                        .foregroundColor(icon.tint)
+
+                case .trailingEdge:
+                    if icon.mirrorIconSpace {
+                        emptyIcon
+                    }
+                    Spacer()
+                    text
+                    Spacer()
```

Comment:
I am surprised that we use this approach. 
`Spacer()` won't work well if we will have a longer text or multiline. Is this component directly from Sourcery?